kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extension/ExtensibleResourceAdapter.java

```diff
@@ -34,8 +34,16 @@
   protected ExtensibleResource<T> resource;
   protected Client client;
 
+  public ExtensibleResourceAdapter() {
+
+  }
+
   public ExtensibleResourceAdapter(ExtensibleResource<T> resource, Client client) {
-    super(resource);
+    init(resource, client);
+  }
+
+  public void init(ExtensibleResource<T> resource, Client client) {
```

Comment:
Why do we need an `init` method vs constructor and why does it need to be public?

Reply:
The previous iteration of this logic did require ExtensibleResourceAdapter subclasses to provide a constructor with those arguments, but it struck me as boilerplate.

It needs to be public because it is called in https://github.com/fabric8io/kubernetes-client/pull/3917/files#diff-4b5b7601ec3d4c3be453677f52bb40c08ebcd44b7879d161139f9484b8eb55a3R43 to inject that state.